jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java

```diff
@@ -42,124 +46,109 @@ public enum ConfigWithImplementations implements Configs.Config {
     // When
     final Integer result = Configs.asInteger("2");
     // Then
     assertThat(result).isEqualTo(2);
   }
 
-  @Test(expected = NumberFormatException.class)
-  public void asIntegerValueWithInvalidStringShouldThrowException() {
-    // When
-    Configs.asInteger("2.15");
-    // Then
-    fail();
+  @Test
+  void asIntegerValueWithInvalidStringShouldThrowException() {
+    assertThrows(NumberFormatException.class, () -> Configs.asInteger("2.15"));
   }
 
   @Test
-  public void asIntegerValueWithNullShouldReturnNull() {
+  void asIntegerValueWithNullShouldReturnNull() {
     // When
     final Integer result = Configs.asInteger(null);
     // Then
     assertThat(result).isNull();
   }
 
   @Test
-  public void getStringValueTest() {
+  void getStringValueTest() {
     String test = RandomStringUtils.randomAlphabetic(10);
-    assertEquals(test, Configs.asString(test));
+    assertThat(Configs.asString(test)).isEqualTo(test);
   }
 
-  @Test
-  public void getFromSystemPropertyWithPropertiesAsFallbackHasKeyInSystemShouldReturnSystemValue() {
+  @DisplayName("System Properties tests")
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("propertyTestData")
+  void propertyTest(String testDesc, String systemKey, String systemValue, String fallbackKey, String fallbackValue, String expected) {
     // Given
     try {
-      System.setProperty("key", "systemValue");
+      System.setProperty(systemKey, systemValue);
       // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
       final Properties fallback = new Properties();
-      fallback.put("key", "fallbackValue");
+      fallback.put(fallbackKey, fallbackValue);
       // When
       final String result = Configs.getFromSystemPropertyWithPropertiesAsFallback(fallback, "key");
       // Then
-      assertThat(result).isEqualTo("systemValue");
+      assertThat(result).isEqualTo(expected);
     } finally {
-      System.clearProperty("key");
+      System.clearProperty(systemKey);
     }
   }
 
-  @Test
-  public void getFromSystemPropertyWithPropertiesAsFallbackHasNotKeyInSystemShouldReturnSystemValue() {
-    // Given
-    try {
-      System.setProperty("not-the-key", "systemValue");
-      // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
-      final Properties fallback = new Properties();
-      fallback.put("key", "fallbackValue");
-      // When
-      final String result = Configs.getFromSystemPropertyWithPropertiesAsFallback(fallback, "key");
-      // Then
-      assertThat(result).isEqualTo("fallbackValue");
-    } finally {
-      System.clearProperty("not-the-key");
-    }
+  @DisplayName("Config interface tests without implementation")
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("configInterfaceWithDefaultsTestData")
+  void configInterfaceWithDefaultsTest(String testDesc, String key, String expected) {
+    assertThat(key).isEqualTo(expected);
```

Comment:
This test is very unclear, the tested method is in the data source (configInterfaceWithDefaultsTestData) and not here. It's really hard to understand what's going on.

You might want to replace the method's signature with something like:
```java
void configInterfaceWithDefaultsTest(String testDesc, ConfigWithDefaults config, String expected) {
```

Also, keeping the method source closer to this one makes it easier to review and t o read in the future